aten/src/ATen/native/SummaryOps.cpp

```diff
@@ -52,10 +52,107 @@ Tensor _bincount_cpu_template(
+// while each bin is inclusive at the lower end and exclusive at the higher,
+// i.e. [start, end) the last bin is inclusive at both, i.e. [start, end], in
+// order to include maxvalue if exists
+template <typename input_t>
+inline int64_t getbin(input_t x, input_t min, input_t max, int64_t nbins) {
+  if (x >= max)
+    return nbins - 1;
+  return static_cast<int64_t>((x - min) * nbins / (max - min));
+}
+
+///////////////// histogram /////////////////
+// This template assumes that input and weights are contiguous (possibly
+// flattened) 1D Tensors of the same size. This can be fixed later on.
+template <typename input_t, typename weights_t>
+std::tuple<Tensor, Tensor> _histogram_cpu_template_uniform_bins(
+    const Tensor& self,
+    int64_t nbins,
+    const Tensor& weights /* optional */,
+    c10::optional<ArrayRef<double>> range,
+    bool density) {
+  // If range is not defined, compute min and max from the values in the Tensor.
+  input_t min;
+  input_t max;
+  if (range.has_value()) {
+    // If range is defined, max must be larger than min.
+    TORCH_CHECK(
+        range.value()[0] < range.value()[1], "max must be larger than min");
+    min = static_cast<input_t>(range.value()[0]);
+    max = static_cast<input_t>(range.value()[1]);
+  } else {
+    min = *self.min().data_ptr<input_t>();
+    max = *self.max().data_ptr<input_t>();
+    // This is done to avoid divide by zero if input min is equal to input max.
+    // In this case computing the histogram can also be skipped altogether, as
+    // it's equal to the sum of weights in the middle bin, and zero everywhere
+    // else.
+    if (min == max) {
+      min -= 1;
+      max += 1;
+    }
+  }
+
+  TORCH_CHECK(nbins > 0, "bins must be > 0");
+  TORCH_CHECK(
+      !(std::isinf(static_cast<double>(min)) ||
+        std::isinf(static_cast<double>(max)) || _isnan(min) || _isnan(max)),
+      "range of [",
+      min,
+      ", ",
+      max,
+      "] is not finite");
+
+  bool has_weights = weights.defined();
+
+  Tensor hist;
+  int64_t self_size = self.size(0);
+
+  const input_t* self_p = self.data_ptr<input_t>();
+  if (has_weights) {
+    hist = native::zeros({nbins}, weights.options());
+    weights_t* output_p = hist.data_ptr<weights_t>();
+    const weights_t* weights_p = weights.data_ptr<weights_t>();
+    // This does the actual work of computing the histogram.
+    // This is single threaded, as other similar operators are single-threaded
+    // in PyTorch today, and a multi-threaded one would be tricky to implement.
+    for (int64_t i = 0; i < self_size; i++) {
+      if (self_p[i] >= min && self_p[i] <= max)
+        output_p[getbin(self_p[i], min, max, nbins)] += weights_p[i];
+    }
+  } else {
+    hist = native::zeros({nbins}, kLong);
+    int64_t* output_p = hist.data_ptr<int64_t>();
+    for (int64_t i = 0; i < self_size; i++) {
+      if (self_p[i] >= min && self_p[i] <= max)
+        output_p[getbin(self_p[i], min, max, nbins)] += 1L;
```

Comment:
I don't think you need to use L here, and it's misleading as L != int64_t. If you want to specify the type of 1 explicitly then you could use a C++-style cast to int64_t.